{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a7a24d1a-81ca-4a0c-b039-abd81552e86e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "os.chdir('..')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "660621ee-26f4-46d3-8a21-1daff21b55aa",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/envs/spacy_gpu/lib/python3.10/site-packages/tqdm/auto.py:22: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n"
     ]
    }
   ],
   "source": [
    "import json\n",
    "import pandas as pd\n",
    "\n",
    "from RE.RelationExtractor import RelationExtractor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "fa60a787-9630-4aa1-9927-bcd8d3737702",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using GPU:  True\n"
     ]
    }
   ],
   "source": [
    "import spacy\n",
    "\n",
    "print('Using GPU: ', spacy.prefer_gpu())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c8736365-a7fc-4cb9-8702-7ce47d30791e",
   "metadata": {},
   "source": [
    "### Read data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "193a860d-9f26-4299-b6e4-dcc9bfd5a5ff",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_path = 'RE/assets/annotations_test.jsonl'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "3d2218aa-4466-4d10-83ff-33d0c7462a6a",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_jsonl(data_path):\n",
    "    with open(data_path, 'r') as f:\n",
    "        json_list = list(f)\n",
    "    data = []\n",
    "    for j in json_list:\n",
    "        data.append(json.loads(j))\n",
    "    return pd.DataFrame(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "8d16a1e6-56cf-461c-9740-f88714de7801",
   "metadata": {},
   "outputs": [],
   "source": [
    "data = read_jsonl(data_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3d8247b7-6d15-4ea7-a345-d741c02c82ca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>text</th>\n",
       "      <th>spans</th>\n",
       "      <th>tokens</th>\n",
       "      <th>relations</th>\n",
       "      <th>answer</th>\n",
       "      <th>meta</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>additionally, whereas incubation of adipose-ti...</td>\n",
       "      <td>[{'id': 35091, 'text': 'adipose-tissue', 'star...</td>\n",
       "      <td>[{'text': 'additionally', 'start': 0, 'end': 1...</td>\n",
       "      <td>[{'head': 8, 'child': 18, 'head_span': {'start...</td>\n",
       "      <td>accept</td>\n",
       "      <td>{'source': 'unkown'}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>in comparison to a pressed echinaceae-preparat...</td>\n",
       "      <td>[{'id': 17727, 'text': 'fec', 'start': 51, 'to...</td>\n",
       "      <td>[{'text': 'in', 'start': 0, 'end': 2, 'id': 0,...</td>\n",
       "      <td>[{'head': 10, 'child': 15, 'head_span': {'star...</td>\n",
       "      <td>accept</td>\n",
       "      <td>{'source': 'unkown'}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>dna damage and replication stress activate th...</td>\n",
       "      <td>[{'id': 35408, 'text': 'dna', 'start': 1, 'tok...</td>\n",
       "      <td>[{'text': ' ', 'start': 0, 'end': 1, 'id': 0, ...</td>\n",
       "      <td>[{'head': 2, 'child': 9, 'head_span': {'start'...</td>\n",
       "      <td>accept</td>\n",
       "      <td>{'source': 'unkown'}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>when compared to controls, the density of 125i...</td>\n",
       "      <td>[{'id': 35071, 'text': '125i-ngf', 'start': 42...</td>\n",
       "      <td>[{'text': 'when', 'start': 0, 'end': 4, 'id': ...</td>\n",
       "      <td>[{'head': 11, 'child': 21, 'head_span': {'star...</td>\n",
       "      <td>accept</td>\n",
       "      <td>{'source': 'unkown'}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>we establish that rnf169 binds to ubiquitylate...</td>\n",
       "      <td>[{'id': 18566, 'text': 'rnf169', 'start': 18, ...</td>\n",
       "      <td>[{'text': 'we', 'start': 0, 'end': 2, 'id': 0,...</td>\n",
       "      <td>[{'head': 4, 'child': 10, 'head_span': {'start...</td>\n",
       "      <td>accept</td>\n",
       "      <td>{'source': 'unkown'}</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                text  \\\n",
       "0  additionally, whereas incubation of adipose-ti...   \n",
       "1  in comparison to a pressed echinaceae-preparat...   \n",
       "2   dna damage and replication stress activate th...   \n",
       "3  when compared to controls, the density of 125i...   \n",
       "4  we establish that rnf169 binds to ubiquitylate...   \n",
       "\n",
       "                                               spans  \\\n",
       "0  [{'id': 35091, 'text': 'adipose-tissue', 'star...   \n",
       "1  [{'id': 17727, 'text': 'fec', 'start': 51, 'to...   \n",
       "2  [{'id': 35408, 'text': 'dna', 'start': 1, 'tok...   \n",
       "3  [{'id': 35071, 'text': '125i-ngf', 'start': 42...   \n",
       "4  [{'id': 18566, 'text': 'rnf169', 'start': 18, ...   \n",
       "\n",
       "                                              tokens  \\\n",
       "0  [{'text': 'additionally', 'start': 0, 'end': 1...   \n",
       "1  [{'text': 'in', 'start': 0, 'end': 2, 'id': 0,...   \n",
       "2  [{'text': ' ', 'start': 0, 'end': 1, 'id': 0, ...   \n",
       "3  [{'text': 'when', 'start': 0, 'end': 4, 'id': ...   \n",
       "4  [{'text': 'we', 'start': 0, 'end': 2, 'id': 0,...   \n",
       "\n",
       "                                           relations  answer  \\\n",
       "0  [{'head': 8, 'child': 18, 'head_span': {'start...  accept   \n",
       "1  [{'head': 10, 'child': 15, 'head_span': {'star...  accept   \n",
       "2  [{'head': 2, 'child': 9, 'head_span': {'start'...  accept   \n",
       "3  [{'head': 11, 'child': 21, 'head_span': {'star...  accept   \n",
       "4  [{'head': 4, 'child': 10, 'head_span': {'start...  accept   \n",
       "\n",
       "                   meta  \n",
       "0  {'source': 'unkown'}  \n",
       "1  {'source': 'unkown'}  \n",
       "2  {'source': 'unkown'}  \n",
       "3  {'source': 'unkown'}  \n",
       "4  {'source': 'unkown'}  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "531f44c7-375c-4aa9-9f8c-31aca857478c",
   "metadata": {},
   "outputs": [],
   "source": [
    "texts = data.text.values.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "2cfc81c7-f74d-4eaf-9762-39fb7c03a2bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "spans = []\n",
    "for k in range(data.shape[0]):\n",
    "    x = data.iloc[k]\n",
    "    spans.append([(x.spans[i]['start'], x.spans[i]['end'], x.spans[i]['label']) for i in range(len(x.spans))])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "f64e2202-d1d4-4be7-9411-958096a12028",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[(36, 50, 'TISSUE'),\n",
       " (93, 121, 'TARGET'),\n",
       " (142, 168, 'TARGET'),\n",
       " (185, 191, 'TARGET'),\n",
       " (326, 345, 'CHEMICAL')]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "spans[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a98b48b1-cf92-4247-b530-4350b739813b",
   "metadata": {},
   "source": [
    "### Predict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "2bf547d5-e3b0-4904-b627-077de778f040",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "if spacy.prefer_gpu():\n",
    "    re = RelationExtractor(model_re='transformer')\n",
    "else:\n",
    "    re = RelationExtractor(model_re='tok2vec')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "6fa337df-50a3-47e2-8ae5-bb298a33cd0f",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  additionally, whereas incubation of adipose-tissue extracts from fed steers did not activate endogenous glycogen synthase (through a presumed phosphoprotein phosphatase mechanism), the enzyme from starved or re-fed (up to 3 days re-feeding) steers was reversibly activated as measured by changes in the value for the a0.5 for glucose 6-phosphate\n",
      "spans: [(5, 'adipose-tissue', 'TISSUE'), (15, 'endogenous glycogen synthase', 'TARGET'), (22, 'phosphoprotein phosphatase', 'TARGET'), (28, 'enzyme', 'TARGET'), (59, 'glucose 6-phosphate', 'CHEMICAL')]\n",
      "entities: ('adipose-tissue', 'endogenous glycogen synthase') --> predicted relation: Activate\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  in comparison to a pressed echinaceae-preparation, fec activated mouse macrophages secrete interleukin-6 and tumor-necrosis-factor and kill protozoa, fungi and bacteria, with higher efficiency\n",
      "spans: [(9, 'fec', 'CHEMICAL'), (12, 'macrophages', 'TISSUE'), (14, 'interleukin-6', 'TARGET'), (16, 'tumor-necrosis-factor', 'TARGET')]\n",
      "entities: ('fec', 'macrophages') --> predicted relation: Activate\n",
      "entities: ('macrophages', 'interleukin-6') --> predicted relation: Express\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :   dna damage and replication stress activate the chk1 signaling pathway, which blocks s phase progression, stabilizes stalled replication forks, and participates in g2 arrest\n",
      "spans: [(1, 'dna', 'TARGET'), (8, 'chk1', 'TARGET')]\n",
      "entities: ('dna', 'chk1') --> predicted relation: Activate\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  when compared to controls, the density of 125i-ngf binding sites was reduced by 30% in the striatum of patients with progressive supranuclear palsy but not reduced in that of patients with parkinson's disease. 125i-ngf binding was not significantly decreased in the nucleus basalis of meynert in either diseases\n",
      "spans: [(8, '125i-ngf', 'TARGET'), (20, 'striatum', 'TISSUE'), (24, 'progressive supranuclear palsy', 'DISEASE'), (35, \"parkinson's disease\", 'DISEASE'), (39, '125i-ngf', 'TARGET'), (49, 'nucleus basalis of meynert', 'TISSUE')]\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  we establish that rnf169 binds to ubiquitylated h2a-lys13/lys15 in a manner that involves its canonical ubiquitin-binding helix and a pair of arginine-rich motifs that interact with the nucleosome acidic patch\n",
      "spans: [(3, 'rnf169', 'TARGET'), (7, 'h2a-lys13', 'TARGET'), (11, 'lys15', 'TARGET'), (19, 'ubiquitin', 'TARGET'), (27, 'arginine', 'TARGET')]\n",
      "entities: ('rnf169', 'h2a-lys13') --> predicted relation: Bind\n",
      "entities: ('rnf169', 'lys15') --> predicted relation: Bind\n",
      "entities: ('rnf169', 'ubiquitin') --> predicted relation: Bind\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :   osteoclast inhibitory lectin (ocil) is a type ii c-type lectin and binds nk cell-associated receptor nkrp1d and sulfated glycosaminoglycans\n",
      "spans: [(1, 'osteoclast inhibitory lectin', 'TARGET'), (5, 'ocil', 'TARGET'), (9, 'type ii c-type lectin', 'TARGET'), (17, 'nk cell-associated receptor', 'TARGET'), (22, 'nkrp1d', 'TARGET'), (24, 'sulfated glycosaminoglycans', 'CHEMICAL')]\n",
      "entities: ('osteoclast inhibitory lectin', 'nk cell-associated receptor') --> predicted relation: Bind\n",
      "entities: ('osteoclast inhibitory lectin', 'nkrp1d') --> predicted relation: Bind\n",
      "entities: ('osteoclast inhibitory lectin', 'sulfated glycosaminoglycans') --> predicted relation: Bind\n",
      "entities: ('ocil', 'nk cell-associated receptor') --> predicted relation: Bind\n",
      "entities: ('ocil', 'nkrp1d') --> predicted relation: Bind\n",
      "entities: ('ocil', 'sulfated glycosaminoglycans') --> predicted relation: Bind\n",
      "entities: ('type ii c-type lectin', 'nk cell-associated receptor') --> predicted relation: Bind\n",
      "entities: ('type ii c-type lectin', 'nkrp1d') --> predicted relation: Bind\n",
      "entities: ('type ii c-type lectin', 'sulfated glycosaminoglycans') --> predicted relation: Bind\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  moreover, beta/h1beta-ko islets had increased hnf1alpha and pdx-1, decreased hnf4 mrna levels, and reduced glucose-stimulated insulin release\n",
      "spans: [(2, 'beta/h1beta', 'TARGET'), (10, 'hnf1alpha', 'TARGET'), (12, 'pdx-1', 'TARGET'), (15, 'hnf4', 'TARGET'), (16, 'mrna', 'TARGET'), (21, 'glucose', 'CHEMICAL'), (24, 'insulin', 'TARGET')]\n",
      "entities: ('beta/h1beta', 'hnf4') --> predicted relation: Decrease\n",
      "entities: ('beta/h1beta', 'mrna') --> predicted relation: Decrease\n",
      "entities: ('beta/h1beta', 'glucose') --> predicted relation: Reduce\n",
      "entities: ('beta/h1beta', 'insulin') --> predicted relation: Reduce\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  conditioned medium (cm) generated from adipose-derived mesenchymal stem cells (mscs) reduces bleomycin (blm)-induced lung injury in rats, suggesting a crucial role in situ of immunomodulatory factors secreted by mscs in these beneficial effects\n",
      "spans: [(7, 'adipose-derived mesenchymal stem cells', 'CELL_LINE'), (14, 'mscs', 'CELL_LINE'), (17, 'bleomycin', 'CHEMICAL'), (20, 'lung injury', 'DISEASE'), (32, 'immunomodulatory factors', 'TARGET'), (36, 'mscs', 'CELL_LINE')]\n",
      "entities: ('adipose-derived mesenchymal stem cells', 'bleomycin') --> predicted relation: Reduce\n",
      "entities: ('adipose-derived mesenchymal stem cells', 'lung injury') --> predicted relation: Reduce\n",
      "entities: ('mscs', 'bleomycin') --> predicted relation: Reduce\n",
      "entities: ('mscs', 'lung injury') --> predicted relation: Reduce\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  ketoconazole is a potent inhibitor of the cytochrome p450 3a (cyp3a) activities and known to inhibit the elimination of drugs metabolized by cyp3a, including budesonide\n",
      "spans: [(0, 'ketoconazole', 'CHEMICAL'), (7, 'cytochrome p450 3a', 'TARGET'), (11, 'cyp3a', 'TARGET'), (24, 'cyp3a', 'TARGET'), (27, 'budesonide', 'CHEMICAL')]\n",
      "entities: ('ketoconazole', 'cytochrome p450 3a') --> predicted relation: Inhibit\n",
      "entities: ('ketoconazole', 'cyp3a') --> predicted relation: Inhibit\n",
      "entities: ('ketoconazole', 'cyp3a') --> predicted relation: Inhibit\n",
      "entities: ('ketoconazole', 'budesonide') --> predicted relation: Inhibit\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  using blue-native gel electrophoresis, we isolated and quantified several types of complexes containing the three snare proteins (syntaxin-1, snap25, vamp), as well as the gabaergic/glutamatergic selectively expressed complexins-i/ii (cplx1/2), in brain tissue homogenates and reconstitution assays with recombinant proteins\n",
      "spans: [(18, 'snare', 'TARGET'), (21, 'syntaxin-1', 'TARGET'), (23, 'snap25', 'TARGET'), (25, 'vamp', 'TARGET'), (37, 'complexins-i', 'TARGET'), (47, 'brain tissue', 'TISSUE')]\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  csk is primarily regulated through its sh2 domain, which is required for membrane translocation of csk via binding to scaffold proteins such as cbp/pag1\n",
      "spans: [(0, 'csk', 'TARGET'), (16, 'csk', 'TARGET'), (24, 'cbp', 'TARGET'), (26, 'pag1', 'TARGET')]\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  furthermore, genetic deletion of ash1l in the mll-af9-transformed cells impaired the maintenance of leukemic cells  and largely blocked the leukemia progression \n",
      "spans: [(5, 'ash1l', 'TARGET'), (8, 'mll-af9-transformed cells', 'CELL_LINE'), (18, 'leukemic cells', 'CELL_LINE'), (25, 'leukemia', 'DISEASE')]\n",
      "entities: ('ash1l', 'leukemia') --> predicted relation: Block\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  simvastatin reduced expression of inos, mmp-1 and -8, rank, and rankl and increased bmp-2 and opg levels in the periodontal tissue\n",
      "spans: [(0, 'simvastatin', 'CHEMICAL'), (4, 'inos', 'TARGET'), (6, 'mmp-1', 'TARGET'), (10, 'rank', 'TARGET'), (13, 'rankl', 'TARGET'), (16, 'bmp-2', 'TARGET'), (18, 'opg', 'TARGET'), (22, 'periodontal tissue', 'TISSUE')]\n",
      "entities: ('simvastatin', 'inos') --> predicted relation: Reduce\n",
      "entities: ('simvastatin', 'mmp-1') --> predicted relation: Reduce\n",
      "entities: ('simvastatin', 'rank') --> predicted relation: Reduce\n",
      "entities: ('simvastatin', 'rankl') --> predicted relation: Reduce\n",
      "entities: ('simvastatin', 'bmp-2') --> predicted relation: Increase\n",
      "entities: ('simvastatin', 'opg') --> predicted relation: Increase\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  both e2 and e2+mpa reduced renal estrogen receptor alpha (erα) and increased renal g protein-coupled estrogen receptor mrna, but neither erα nor erß protein was different between groups\n",
      "spans: [(1, 'e2', 'CHEMICAL'), (3, 'e2+mpa', 'CHEMICAL'), (5, 'renal estrogen receptor alpha', 'TARGET'), (10, 'erα', 'TARGET'), (15, 'g protein', 'TARGET'), (19, 'estrogen receptor', 'TARGET'), (21, 'mrna', 'TARGET'), (25, 'erα', 'TARGET'), (27, 'erß protein', 'TARGET')]\n",
      "entities: ('e2', 'renal estrogen receptor alpha') --> predicted relation: Reduce\n",
      "entities: ('e2', 'erα') --> predicted relation: Reduce\n",
      "entities: ('e2', 'g protein') --> predicted relation: Increase\n",
      "entities: ('e2', 'estrogen receptor') --> predicted relation: Increase\n",
      "entities: ('e2', 'mrna') --> predicted relation: Increase\n",
      "entities: ('e2+mpa', 'renal estrogen receptor alpha') --> predicted relation: Reduce\n",
      "entities: ('e2+mpa', 'erα') --> predicted relation: Reduce\n",
      "entities: ('e2+mpa', 'g protein') --> predicted relation: Increase\n",
      "entities: ('e2+mpa', 'estrogen receptor') --> predicted relation: Increase\n",
      "entities: ('e2+mpa', 'mrna') --> predicted relation: Increase\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  rassf6 binds mdm2 and regulates p53 expression\n",
      "spans: [(0, 'rassf6', 'TARGET'), (2, 'mdm2', 'TARGET'), (5, 'p53', 'TARGET')]\n",
      "entities: ('rassf6', 'mdm2') --> predicted relation: Bind\n",
      "entities: ('rassf6', 'p53') --> predicted relation: Regulate\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  despite these beneficial effects, rapamycin-treated lean mice showed greater glucose intolerance, reduced insulin sensitivity, lower muscle glut4 expression and changes in chromium levels in tissues even with high insulin levels\n",
      "spans: [(5, 'rapamycin', 'CHEMICAL'), (12, 'glucose', 'CHEMICAL'), (16, 'insulin', 'TARGET'), (21, 'glut4', 'TARGET'), (26, 'chromium', 'CHEMICAL'), (33, 'insulin', 'TARGET')]\n",
      "entities: ('rapamycin', 'insulin') --> predicted relation: Reduce\n",
      "entities: ('rapamycin', 'glut4') --> predicted relation: Decrease\n",
      "entities: ('rapamycin', 'chromium') --> predicted relation: Decrease\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  in the absence of leptin, cla significantly depleted adiponectin and induced insulin resistance, but it did not increase hepatic triglyceride concentrations or adipose inflammation, marked by interleukin-6 and tumor necrosis factor-alpha mrna expression\n",
      "spans: [(4, 'leptin', 'TARGET'), (6, 'cla', 'CHEMICAL'), (9, 'adiponectin', 'TARGET'), (12, 'insulin', 'TARGET'), (20, 'hepatic triglyceride', 'TARGET'), (29, 'interleukin-6', 'TARGET'), (31, 'tumor necrosis factor-alpha', 'TARGET'), (36, 'mrna', 'TARGET')]\n",
      "entities: ('cla', 'insulin') --> predicted relation: Induce\n",
      "entities: ('cla', 'hepatic triglyceride') --> predicted relation: Increase\n",
      "entities: ('cla', 'interleukin-6') --> predicted relation: Increase\n",
      "entities: ('cla', 'tumor necrosis factor-alpha') --> predicted relation: Increase\n",
      "entities: ('cla', 'mrna') --> predicted relation: Increase\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  the present invention relates to a kind of cells, it is characterized in that, the exogenous receptor and exogenous il-18r binding protein of the cell expression specificity combination target antigen, the exogenous il-18r binding protein can specifically bind il-18r and enhancing il-18r activity, and the target antigen includes tumour antigen or pathogen antigen\n",
      "spans: [(17, 'exogenous receptor', 'TARGET'), (20, 'exogenous il-18r binding protein', 'TARGET'), (34, 'exogenous il-18r binding protein', 'TARGET'), (41, 'il-18r', 'TARGET'), (44, 'il-18r', 'TARGET'), (50, 'antigen', 'TARGET'), (52, 'tumour', 'DISEASE'), (53, 'antigen', 'TARGET'), (55, 'pathogen', 'TARGET'), (56, 'antigen', 'TARGET')]\n",
      "entities: ('exogenous il-18r binding protein', 'il-18r') --> predicted relation: Bind\n",
      "entities: ('exogenous il-18r binding protein', 'il-18r') --> predicted relation: Bind\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  in contrast, the proliferative response to a pepsin-extracted type 5 m-protein from streptococcus pyogenes (pep m5), a recently identified superantigen, required signals provided by phorbol 12-myristate 13-acetate (pma), il-1 and il-6. pep m5 alone did not induce il-2r expression; however, when combined with pma, il-1 and il-6, il-2r was expressed\n",
      "spans: [(8, 'pepsin-extracted type 5 m-protein', 'TARGET'), (20, 'pep m5', 'TARGET'), (27, 'superantigen', 'TARGET'), (33, 'phorbol 12-myristate 13-acetate', 'CHEMICAL'), (41, 'pma', 'CHEMICAL'), (44, 'il-1', 'TARGET'), (46, 'il-6', 'TARGET'), (48, 'pep m5', 'TARGET'), (54, 'il-2r', 'TARGET'), (62, 'pma', 'CHEMICAL'), (64, 'il-1', 'TARGET'), (66, 'il-6', 'TARGET'), (68, 'il-2r', 'TARGET')]\n",
      "entities: ('pep m5', 'il-2r') --> predicted relation: Induce\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  we have previously shown that human periodontal ligament cells express estrogen receptor beta (erbeta) but not eralpha, whereas human breast cancer cells (mcf7) express both eralpha and erbeta\n",
      "spans: [(5, 'human periodontal ligament cells', 'CELL_LINE'), (10, 'estrogen receptor beta (', 'TARGET'), (14, 'erbeta', 'TARGET'), (18, 'eralpha', 'TARGET'), (21, 'human breast cancer cells', 'CELL_LINE'), (26, 'mcf7', 'CELL_LINE'), (30, 'eralpha', 'TARGET'), (32, 'erbeta', 'TARGET')]\n",
      "entities: ('human periodontal ligament cells', 'estrogen receptor beta (') --> predicted relation: Express\n",
      "entities: ('human periodontal ligament cells', 'erbeta') --> predicted relation: Express\n",
      "entities: ('human periodontal ligament cells', 'eralpha') --> predicted relation: Express\n",
      "entities: ('human periodontal ligament cells', 'eralpha') --> predicted relation: Express\n",
      "entities: ('human periodontal ligament cells', 'erbeta') --> predicted relation: Express\n",
      "entities: ('human breast cancer cells', 'estrogen receptor beta (') --> predicted relation: Express\n",
      "entities: ('human breast cancer cells', 'erbeta') --> predicted relation: Express\n",
      "entities: ('human breast cancer cells', 'eralpha') --> predicted relation: Express\n",
      "entities: ('human breast cancer cells', 'eralpha') --> predicted relation: Express\n",
      "entities: ('human breast cancer cells', 'erbeta') --> predicted relation: Express\n",
      "entities: ('mcf7', 'estrogen receptor beta (') --> predicted relation: Express\n",
      "entities: ('mcf7', 'erbeta') --> predicted relation: Express\n",
      "entities: ('mcf7', 'eralpha') --> predicted relation: Express\n",
      "entities: ('mcf7', 'eralpha') --> predicted relation: Express\n",
      "entities: ('mcf7', 'erbeta') --> predicted relation: Express\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :   calponin isolated from chicken gizzard smooth muscle binds in vitro to actin in a ca(2+)-independent manner and thereby inhibits the actin-activated mg(2+)-adenosinetriphosphatase of smooth muscle myosin\n",
      "spans: [(1, 'calponin', 'TARGET'), (12, 'actin', 'TARGET'), (26, 'smooth muscle', 'TISSUE'), (28, 'myosin', 'TARGET')]\n",
      "entities: ('calponin', 'actin') --> predicted relation: Bind\n",
      "entities: ('calponin', 'smooth muscle') --> predicted relation: Inhibit\n",
      "entities: ('calponin', 'myosin') --> predicted relation: Inhibit\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  amidase synthesis is not regulated in the same way as histidase synthesis by the availability of other nitrogen compounds for growth\n",
      "spans: [(0, 'amidase', 'TARGET'), (10, 'histidase', 'TARGET'), (17, 'nitrogen', 'CHEMICAL')]\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  in the ventral lumbar cord there was a comparable decrease of trh and 5-ht (-80%) and a smaller but significant decrease in the nucleus accumbens (-55%) and septal nuclei (-38%). p-chlorophenylalanine (pcpa; 250 mg/kg x 2) reduced levels of 5-ht (-80%), without significantly altering those of noradrenaline. p-chlorophenylalanine also significantly reduced levels of trh in the nucleus accumbens (-72%) but not in the other regions of brain or spinal cord taken\n",
      "spans: [(11, 'trh', 'TARGET'), (13, '5-ht', 'TARGET'), (42, 'p-chlorophenylalanine', 'CHEMICAL'), (46, 'pcpa', 'CHEMICAL'), (58, '5-ht', 'TARGET'), (71, 'noradrenaline', 'CHEMICAL'), (73, 'p-chlorophenylalanine', 'CHEMICAL'), (81, 'trh', 'TARGET'), (97, 'brain', 'TISSUE'), (99, 'spinal', 'TISSUE')]\n",
      "entities: ('p-chlorophenylalanine', 'trh') --> predicted relation: Decrease\n",
      "entities: ('p-chlorophenylalanine', '5-ht') --> predicted relation: Decrease\n",
      "entities: ('p-chlorophenylalanine', '5-ht') --> predicted relation: Reduce\n",
      "entities: ('p-chlorophenylalanine', 'trh') --> predicted relation: Reduce\n",
      "entities: ('pcpa', 'trh') --> predicted relation: Decrease\n",
      "entities: ('pcpa', '5-ht') --> predicted relation: Decrease\n",
      "entities: ('pcpa', '5-ht') --> predicted relation: Reduce\n",
      "entities: ('pcpa', 'trh') --> predicted relation: Reduce\n",
      "entities: ('p-chlorophenylalanine', 'trh') --> predicted relation: Decrease\n",
      "entities: ('p-chlorophenylalanine', '5-ht') --> predicted relation: Decrease\n",
      "entities: ('p-chlorophenylalanine', '5-ht') --> predicted relation: Reduce\n",
      "entities: ('p-chlorophenylalanine', 'trh') --> predicted relation: Reduce\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  previous work has shown that a2re rnas are transported to myelin in oligodendrocytes and to dendrites in neurons. hnrnp e1 is an rna-binding protein that regulates translation of specific mrnas\n",
      "spans: [(5, 'a2re rnas', 'TARGET'), (10, 'myelin', 'TARGET'), (12, 'oligodendrocytes', 'TISSUE'), (15, 'dendrites', 'TISSUE'), (17, 'neurons', 'TISSUE'), (19, 'hnrnp e1', 'TARGET'), (23, 'rna', 'TARGET'), (32, 'mrnas', 'TARGET')]\n",
      "entities: ('hnrnp e1', 'mrnas') --> predicted relation: Regulate\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :   antagonist anti-cd28 antibodies prevent t-cell costimulation and are functionally different from ctla4ig since they cannot block ctla-4 and pdl-1 co-inhibitory signals\n",
      "spans: [(4, 'cd28', 'TARGET'), (5, 'antibodies', 'TARGET'), (7, 't-cell', 'CELL_LINE'), (16, 'ctla4ig', 'TARGET'), (22, 'ctla-4', 'TARGET'), (24, 'pdl-1', 'TARGET')]\n",
      "entities: ('cd28', 'ctla-4') --> predicted relation: Block\n",
      "entities: ('cd28', 'pdl-1') --> predicted relation: Block\n",
      "entities: ('antibodies', 'ctla-4') --> predicted relation: Block\n",
      "entities: ('antibodies', 'pdl-1') --> predicted relation: Block\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  the transcription factor nf-kappa b was not activated following ncp bvdv infection, but ncp bvdv infection was not able to block the activation of nf-kappa b by either sfv or tumor necrosis factor alpha\n",
      "spans: [(1, 'transcription factor nf-kappa b', 'TARGET'), (12, 'bvdv infection', 'DISEASE'), (17, 'bvdv infection', 'DISEASE'), (27, 'nf-kappa b', 'TARGET'), (33, 'sfv', 'TARGET'), (35, 'tumor necrosis factor alpha', 'TARGET')]\n",
      "entities: ('bvdv infection', 'transcription factor nf-kappa b') --> predicted relation: Activate\n",
      "entities: ('bvdv infection', 'nf-kappa b') --> predicted relation: Block\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  s100a8/a9-mcam binding activates mitogen-activated protein kinase kinase kinase 8 (map3k8), also termed tpl2, leading to strong activation of the transcription factor etv4 and subsequent induction of matrix metalloproteinase-25 (mmp25), and finally to induction of melanoma lung tropic metastasis\n",
      "spans: [(0, 's100a8', 'TARGET'), (4, 'mcam', 'TARGET'), (7, 'mitogen-activated protein kinase kinase kinase 8', 'TARGET'), (16, 'map3k8', 'TARGET'), (21, 'tpl2', 'TARGET'), (29, 'transcription factor', 'TARGET'), (31, 'etv4', 'TARGET'), (36, 'matrix metalloproteinase-25', 'TARGET'), (39, 'mmp25', 'TARGET'), (47, 'melanoma lung tropic metastasis', 'DISEASE')]\n",
      "entities: ('s100a8', 'mitogen-activated protein kinase kinase kinase 8') --> predicted relation: Activate\n",
      "entities: ('s100a8', 'map3k8') --> predicted relation: Activate\n",
      "entities: ('s100a8', 'tpl2') --> predicted relation: Activate\n",
      "entities: ('s100a8', 'transcription factor') --> predicted relation: Activate\n",
      "entities: ('s100a8', 'etv4') --> predicted relation: Activate\n",
      "entities: ('mcam', 'mitogen-activated protein kinase kinase kinase 8') --> predicted relation: Activate\n",
      "entities: ('mcam', 'map3k8') --> predicted relation: Activate\n",
      "entities: ('mcam', 'tpl2') --> predicted relation: Activate\n",
      "entities: ('mcam', 'transcription factor') --> predicted relation: Activate\n",
      "entities: ('mcam', 'etv4') --> predicted relation: Activate\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  it is also known that the mitogen-activated protein kinase sty1 negatively regulates gcn2 and hri2 activities under oxidative stress\n",
      "spans: [(6, 'mitogen-activated protein kinase', 'TARGET'), (11, 'sty1', 'TARGET'), (14, 'gcn2', 'TARGET'), (16, 'hri2', 'TARGET')]\n",
      "entities: ('mitogen-activated protein kinase', 'gcn2') --> predicted relation: Regulate\n",
      "entities: ('mitogen-activated protein kinase', 'hri2') --> predicted relation: Regulate\n",
      "entities: ('sty1', 'gcn2') --> predicted relation: Regulate\n",
      "entities: ('sty1', 'hri2') --> predicted relation: Regulate\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  we show that thrombin mediated activation of human platelets causes binding of grk5 to par-1 and that deletion of the mouse homologue grk5 enhances thrombin induced platelet activation sensitivity and increases platelet accumulation at the site of vascular injury\n",
      "spans: [(3, 'thrombin', 'TARGET'), (8, 'platelets', 'TISSUE'), (12, 'grk5', 'TARGET'), (14, 'par-1', 'TARGET'), (22, 'grk5', 'TARGET'), (24, 'thrombin', 'TARGET'), (26, 'platelet', 'TISSUE'), (31, 'platelet', 'TISSUE'), (37, 'vascular injury', 'DISEASE')]\n",
      "entities: ('grk5', 'platelet') --> predicted relation: Increase\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  however, active 1-34 rat pth decreased phosphate to the same level as vehicle-treated rats and increased 1,25(oh)(2)d to very high levels, whereas active 1-84 pth decreased phosphate but maintained 1,25(oh)(2)d\n",
      "spans: [(7, 'pth', 'TARGET'), (9, 'phosphate', 'CHEMICAL'), (21, '1,25(oh)(2)d', 'CHEMICAL'), (32, 'pth', 'TARGET'), (34, 'phosphate', 'CHEMICAL'), (37, '1,25(oh)(2)d', 'CHEMICAL')]\n",
      "entities: ('pth', 'phosphate') --> predicted relation: Decrease\n",
      "entities: ('pth', '1,25(oh)(2)d') --> predicted relation: Increase\n",
      "entities: ('pth', 'phosphate') --> predicted relation: Decrease\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  likewise, treatment of w256 cells with various lox but not cox inhibitors induced apoptotic cell death, which could be partially inhibited by exogenous 12(s)- or 15(s)-hydroxyeicosatetraenoic acids\n",
      "spans: [(4, 'w256 cells', 'CELL_LINE'), (8, 'lox', 'TARGET'), (11, 'cox inhibitors', 'TARGET'), (14, 'apoptotic', 'PATHWAY'), (27, '15(s)-hydroxyeicosatetraenoic acids', 'CHEMICAL')]\n",
      "entities: ('w256 cells', 'apoptotic') --> predicted relation: Induce\n",
      "entities: ('lox', 'apoptotic') --> predicted relation: Induce\n",
      "entities: ('cox inhibitors', 'apoptotic') --> predicted relation: Induce\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  in addition, insulin did not activate the related protein kinase jnk, demonstrating the specificity of insulin for the erk pathway\n",
      "spans: [(3, 'insulin', 'TARGET'), (9, 'protein kinase', 'TARGET'), (11, 'jnk', 'TARGET'), (17, 'insulin', 'TARGET'), (20, 'erk pathway', 'PATHWAY')]\n",
      "entities: ('insulin', 'protein kinase') --> predicted relation: Activate\n",
      "entities: ('insulin', 'jnk') --> predicted relation: Activate\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  anti-urokinase-type plasminogen activator (upa) and anti-upar antibodies reduced the migration and invasion activities of r-1 and r-2 cells to baseline levels\n",
      "spans: [(0, 'anti-urokinase-type plasminogen activator', 'TARGET'), (8, 'upa', 'TARGET'), (11, 'anti-upar antibodies', 'TARGET'), (22, 'r-1', 'CELL_LINE'), (24, 'r-2 cells', 'CELL_LINE')]\n",
      "entities: ('anti-urokinase-type plasminogen activator', 'r-1') --> predicted relation: Reduce\n",
      "entities: ('anti-urokinase-type plasminogen activator', 'r-2 cells') --> predicted relation: Reduce\n",
      "entities: ('upa', 'r-1') --> predicted relation: Reduce\n",
      "entities: ('upa', 'r-2 cells') --> predicted relation: Reduce\n",
      "entities: ('anti-upar antibodies', 'r-1') --> predicted relation: Reduce\n",
      "entities: ('anti-upar antibodies', 'r-2 cells') --> predicted relation: Reduce\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :   in our search for natural soluble epoxide hydrolase (seh) inhibitors from plants, we found that the methanolic extract of the leaves and twigs of archidendron clypearia (jack.) nielsen (fabaceae) significantly inhibits seh in vitro\n",
      "spans: [(5, 'natural soluble epoxide hydrolase', 'TARGET'), (10, 'seh', 'TARGET'), (20, 'methanolic', 'CHEMICAL'), (40, 'seh', 'TARGET')]\n",
      "entities: ('methanolic', 'seh') --> predicted relation: Inhibit\n",
      "entities: ('methanolic', 'seh') --> predicted relation: Inhibit\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  reducing low-density lipoprotein receptor-related proteins (lrp) 5 and lrp6 protein expression prevented wnt3a-induced inactivation of nfatc1; however, deletion of β-catenin did not block wnt3a inactivation of nfatc1, suggesting that this effect was mediated by a noncanonical pathway\n",
      "spans: [(1, 'low-density lipoprotein receptor-related proteins', 'TARGET'), (9, '(lrp) 5', 'TARGET'), (14, 'lrp6', 'TARGET'), (18, 'wnt3a', 'TARGET'), (23, 'nfatc1', 'TARGET'), (29, 'β-catenin', 'TARGET'), (35, 'wnt3a', 'TARGET'), (38, 'nfatc1', 'TARGET')]\n",
      "entities: ('β-catenin', 'wnt3a') --> predicted relation: Block\n",
      "entities: ('β-catenin', 'nfatc1') --> predicted relation: Block\n",
      "entities: ('β-catenin', 'wnt3a') --> predicted relation: Block\n",
      "entities: ('β-catenin', 'nfatc1') --> predicted relation: Block\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  since these ventricular olig2-positive cells do not express the oligodendrocyte marker sox10, it is not clear whether these cells indeed belong to the oligodendrocyte lineage\n",
      "spans: [(2, 'ventricular olig2-positive cells', 'CELL_LINE'), (11, 'oligodendrocyte', 'TISSUE'), (13, 'sox10', 'TARGET'), (26, 'oligodendrocyte', 'TISSUE')]\n",
      "entities: ('ventricular olig2-positive cells', 'oligodendrocyte') --> predicted relation: Express\n",
      "entities: ('ventricular olig2-positive cells', 'sox10') --> predicted relation: Express\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  here we report that the up-regulation of n-methyl-d-aspartate receptors (nmdars) induced by expression of constitutively active neuronal src (n-src), in which the c-terminus tyrosine is mutated to phenylalanine (n-src/y535f), is significantly reduced by dysfunctions of the sh2 and/or sh3 domains of the protein\n",
      "spans: [(9, 'n-methyl-d-aspartate', 'TARGET'), (18, 'nmdars', 'TARGET'), (26, 'neuronal src', 'TARGET'), (29, 'n-src', 'TARGET'), (37, 'c-terminus tyrosine', 'TARGET'), (44, 'phenylalanine', 'TARGET'), (46, 'n-src', 'TARGET'), (50, 'y535f', 'TARGET'), (60, 'sh2', 'CHEMICAL'), (62, 'sh3', 'CHEMICAL')]\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  in the synchronous subgroup (stage iv tumors), 57% of the primary tumor and metastases pairs did not express p27 protein and the remainder were low expressors\n",
      "spans: [(7, 'tumors', 'DISEASE'), (15, 'tumor', 'DISEASE'), (17, 'metastases', 'DISEASE'), (22, 'p27', 'TARGET')]\n",
      "entities: ('tumor', 'p27') --> predicted relation: Express\n",
      "entities: ('metastases', 'p27') --> predicted relation: Express\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  on the other hand, we have observed that, whereas proto-vav can slightly elevate jnk/sapk activity, oncogenic vav potently activates jnk/sapk to an extent comparable to that elicited by two guanine-nucleotide exchange factors for rho family members, dbl and ost\n",
      "spans: [(11, 'proto-vav', 'TARGET'), (17, 'jnk', 'TARGET'), (19, 'sapk', 'TARGET'), (22, 'oncogenic vav', 'TARGET'), (26, 'jnk', 'TARGET'), (28, 'sapk', 'TARGET'), (38, 'guanine-nucleotide exchange factors', 'TARGET'), (44, 'rho', 'TARGET'), (48, 'dbl', 'TARGET'), (50, 'ost', 'TARGET')]\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  the enzyme nitric oxide synthase (nos) present in neurons is activated by ca2+ influx associated with activation of glutamate receptors\n",
      "spans: [(2, 'nitric oxide synthase', 'TARGET'), (6, 'nos', 'TARGET'), (10, 'neurons', 'TISSUE'), (14, 'ca2', 'CHEMICAL'), (21, 'glutamate', 'CHEMICAL')]\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  the enzyme was not inhibited by the mammalian ace inhibitors lisinopril and enalaprilat\n",
      "spans: [(1, 'enzyme', 'TARGET'), (7, 'mammalian ace inhibitors', 'TARGET'), (10, 'lisinopril', 'CHEMICAL'), (12, 'enalaprilat', 'CHEMICAL')]\n",
      "entities: ('lisinopril', 'enzyme') --> predicted relation: Inhibit\n",
      "entities: ('enalaprilat', 'enzyme') --> predicted relation: Inhibit\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  conversely, knockdown of grim-19 induced aberrant stat3 activation and accelerated gc cell growth in vitro and in vivo, and this could be partly attenuated by the blockage of stat3 activation\n",
      "spans: [(4, 'grim-19', 'TARGET'), (7, 'stat3', 'TARGET'), (11, 'gc cell', 'CELL_LINE'), (30, 'stat3', 'TARGET')]\n",
      "entities: ('grim-19', 'stat3') --> predicted relation: Activate\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  we found that transfection of expression vectors containing usf-1 or usf-2 cdnas activated cyp1a2 reporter gene activity, while a dominant-negative usf-2 expression vector blocked such activity\n",
      "spans: [(8, 'usf-1', 'TARGET'), (10, 'usf-2', 'TARGET'), (11, 'cdnas', 'TARGET'), (13, 'cyp1a2', 'TARGET'), (23, 'usf-2', 'TARGET')]\n",
      "entities: ('usf-1', 'cyp1a2') --> predicted relation: Activate\n",
      "entities: ('usf-2', 'cyp1a2') --> predicted relation: Activate\n",
      "entities: ('cdnas', 'cyp1a2') --> predicted relation: Activate\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  c7h2 induces β-actin polymerization and f-actin stabilization, linked with abundant generation of superoxide anions and apoptosis\n",
      "spans: [(0, 'c7h2', 'CHEMICAL'), (2, 'β-actin', 'TARGET'), (7, 'f-actin', 'TARGET'), (20, 'apoptosis', 'PATHWAY')]\n",
      "entities: ('c7h2', 'β-actin') --> predicted relation: Induce\n",
      "entities: ('c7h2', 'f-actin') --> predicted relation: Induce\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  kcne2 is also expressed in human heart and can associate with kcnq1 to suppress its current amplitude and slow the deactivation gating process\n",
      "spans: [(0, 'kcne2', 'TARGET'), (6, 'heart', 'TISSUE'), (11, 'kcnq1', 'TARGET')]\n",
      "entities: ('kcne2', 'heart') --> predicted relation: Express\n",
      "entities: ('kcne2', 'kcnq1') --> predicted relation: Bind\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  mechanistically, galpha9 functions rapidly following receptor stimulation to negatively regulate pi3k/pten, adenylyl cyclase, and guanylyl cyclase pathways\n",
      "spans: [(2, 'galpha9', 'TARGET'), (11, 'pi3k', 'TARGET'), (13, 'pten', 'TARGET'), (15, 'adenylyl cyclase', 'TARGET'), (19, 'guanylyl cyclase', 'TARGET')]\n",
      "entities: ('galpha9', 'pi3k') --> predicted relation: Regulate\n",
      "entities: ('galpha9', 'pten') --> predicted relation: Regulate\n",
      "entities: ('galpha9', 'adenylyl cyclase') --> predicted relation: Regulate\n",
      "entities: ('galpha9', 'guanylyl cyclase') --> predicted relation: Regulate\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  gnrh, a ca2+ mobilizing ligand, binds to its respective binding protein, which is a member of the seven transmembrane domain receptor family and activates a g-protein (gq). 2\n",
      "spans: [(0, 'gnrh', 'TARGET'), (3, 'ca2', 'TARGET'), (22, 'transmembrane domain receptor', 'TARGET'), (29, 'g-protein', 'TARGET')]\n",
      "entities: ('gnrh', 'g-protein') --> predicted relation: Activate\n",
      "entities: ('ca2', 'g-protein') --> predicted relation: Activate\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  the increase in eosinophils was blocked by pretreatment of animals with an anti-il5 monoclonal antibody (trfk5) in doses similar to those which inhibit eosinophilia in guinea pig asthma models\n",
      "spans: [(3, 'eosinophils', 'TISSUE'), (12, 'anti-il5 monoclonal antibody', 'TARGET'), (18, 'trfk5', 'TARGET'), (27, 'eosinophilia', 'DISEASE'), (31, 'asthma', 'DISEASE')]\n",
      "entities: ('anti-il5 monoclonal antibody', 'eosinophils') --> predicted relation: Block\n",
      "entities: ('anti-il5 monoclonal antibody', 'eosinophilia') --> predicted relation: Inhibit\n",
      "entities: ('trfk5', 'eosinophils') --> predicted relation: Block\n",
      "entities: ('trfk5', 'eosinophilia') --> predicted relation: Inhibit\n",
      "entities: ('trfk5', 'asthma') --> predicted relation: Inhibit\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  under similar conditions, catechin and epicatechin fail to inhibit the hydroperoxide (hpx) response to tpa whereas the procyanidin dimers inhibit this response by almost 40%\n",
      "spans: [(4, 'catechin', 'CHEMICAL'), (6, 'epicatechin', 'CHEMICAL'), (11, 'hydroperoxide', 'CHEMICAL'), (13, 'hpx', 'CHEMICAL'), (17, 'tpa', 'TARGET'), (20, 'procyanidin', 'CHEMICAL')]\n",
      "entities: ('catechin', 'hydroperoxide') --> predicted relation: Inhibit\n",
      "entities: ('catechin', 'hpx') --> predicted relation: Inhibit\n",
      "entities: ('catechin', 'tpa') --> predicted relation: Inhibit\n",
      "entities: ('epicatechin', 'hydroperoxide') --> predicted relation: Inhibit\n",
      "entities: ('epicatechin', 'hpx') --> predicted relation: Inhibit\n",
      "entities: ('epicatechin', 'tpa') --> predicted relation: Inhibit\n",
      "entities: ('procyanidin', 'hydroperoxide') --> predicted relation: Inhibit\n",
      "entities: ('procyanidin', 'hpx') --> predicted relation: Inhibit\n",
      "entities: ('procyanidin', 'tpa') --> predicted relation: Inhibit\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  induction of tnf from lgl did not require the participation of monocytes or t cells because treatment of the lgl with cd14 or cd15 to eliminate contaminating monocytes and cd3, cd4, or cd8 to eliminate contaminating t cells did not decrease the level of tnf produced from the treated lgl\n",
      "spans: [(2, 'tnf', 'TARGET'), (4, 'lgl', 'DISEASE'), (11, 'monocytes', 'TISSUE'), (13, 't cells', 'TISSUE'), (19, 'lgl', 'DISEASE'), (21, 'cd14', 'TARGET'), (23, 'cd15', 'TARGET'), (27, 'monocytes', 'TISSUE'), (29, 'cd3', 'TARGET'), (31, 'cd4', 'TARGET'), (34, 'cd8', 'TARGET'), (38, 't cells', 'TISSUE'), (46, 'tnf', 'TARGET'), (51, 'lgl', 'DISEASE')]\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :      saposin c, in a cell type-specific manner, upregulates upa/upar and immediate early gene c-jun expression, stimulates cell proliferation, migration and invasion and activates p42/44 and sapk/jnk mapk pathways in prostate stromal and cancer cells\n",
      "spans: [(1, 'saposin c', 'TARGET'), (13, 'upa', 'TARGET'), (15, 'upar', 'TARGET'), (20, 'c-jun', 'TARGET'), (36, 'sapk', 'TARGET'), (38, 'jnk', 'TARGET'), (39, 'mapk', 'TARGET'), (42, 'prostate stromal and cancer cells', 'TISSUE')]\n",
      "entities: ('saposin c', 'upa') --> predicted relation: Regulate\n",
      "entities: ('saposin c', 'upar') --> predicted relation: Regulate\n",
      "entities: ('saposin c', 'c-jun') --> predicted relation: Regulate\n",
      "entities: ('saposin c', 'sapk') --> predicted relation: Activate\n",
      "entities: ('saposin c', 'jnk') --> predicted relation: Activate\n",
      "entities: ('saposin c', 'mapk') --> predicted relation: Activate\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  we have reported that expression of connective tissue growth factor (ctgf) is increased in the myocardium of diabetic mice\n",
      "spans: [(6, 'connective tissue growth factor', 'TARGET'), (11, 'ctgf', 'TARGET'), (17, 'myocardium', 'TISSUE'), (19, 'diabetic', 'DISEASE')]\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  the plateau phase of ca2+ was inhibited competitively by mg2+ (0.5--50 mm) and non-competitively by mn2+ (30 micrometer--1 mm), whereas the maximal contraction of ca2+ was not inhibited by either ion\n",
      "spans: [(4, 'ca2', 'TARGET'), (10, 'mg2', 'CHEMICAL'), (23, 'mn2+', 'CHEMICAL'), (36, 'ca2', 'TARGET')]\n",
      "entities: ('mg2', 'ca2') --> predicted relation: Inhibit\n",
      "entities: ('mg2', 'ca2') --> predicted relation: Inhibit\n",
      "entities: ('mn2+', 'ca2') --> predicted relation: Inhibit\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  furthermore, it has been shown that, in contrast to thrombin, which induces proteolysis of other proteins besides its receptor, the thrombin receptor agonist peptide, serine-phenylalanine-leucine-leucine-arginine-asparagine-proline (sfllrnp), is only known to activate this receptor\n",
      "spans: [(11, 'thrombin', 'TARGET'), (15, 'proteolysis', 'PATHWAY'), (24, 'thrombin receptor', 'TARGET'), (27, 'peptide, serine-phenylalanine-leucine-leucine-arginine-asparagine-proline', 'CHEMICAL'), (43, 'sfllrnp', 'CHEMICAL')]\n",
      "entities: ('thrombin', 'proteolysis') --> predicted relation: Induce\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  we also show how vascular endothelial growth factor (vegf) regulates prkcb promoter function in cll cells, stimulating pkcβ gene transcription via increased association of sp1 and decreased association of stat3\n",
      "spans: [(4, 'vascular endothelial growth factor', 'TARGET'), (9, 'vegf', 'TARGET'), (12, 'prkcb', 'TARGET'), (16, 'cll', 'DISEASE'), (20, 'pkcβ', 'TARGET'), (27, 'sp1', 'TARGET'), (32, 'stat3', 'TARGET')]\n",
      "entities: ('vascular endothelial growth factor', 'prkcb') --> predicted relation: Regulate\n",
      "entities: ('vegf', 'prkcb') --> predicted relation: Regulate\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  chk1 phosphorylates cdc25c on serine 216 and inactivates it whereas cdc25c dephosphorylates tyrosine 15 phosphate of cdc2 and activates the cdc2-cyclin b complex\n",
      "spans: [(0, 'chk1', 'TARGET'), (2, 'cdc25c', 'TARGET'), (4, 'serine 216', 'TARGET'), (10, 'cdc25c', 'TARGET'), (12, 'tyrosine 15 phosphate', 'CHEMICAL'), (16, 'cdc2', 'TARGET'), (20, 'cdc2', 'TARGET'), (22, 'cyclin b', 'TARGET')]\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Text :  we found that fgl2 gene silencing inhibits apoptosis and improves heart function of streptozotocin (stz)-induced diabetes rats, the possible mechanism maybe that fgl2 gene silencing reduces the tumour necrosis factor (tnf)±levels, decreases the expression of b-cell lymphoma-2 (bcl2), bcl-2-associated x (bax), toll-like receptors 4 (tlr4) and p38 mitogen-activated protein kinase (mapk)\n",
      "spans: [(3, 'fgl2', 'TARGET'), (7, 'apoptosis', 'PATHWAY'), (10, 'heart', 'TISSUE'), (13, 'streptozotocin', 'CHEMICAL'), (16, 'diabetes', 'DISEASE'), (24, 'fgl2', 'TARGET'), (29, 'tumour necrosis factor', 'TARGET'), (39, 'b-cell lymphoma-2', 'TARGET'), (44, 'bcl2', 'TARGET'), (47, 'bcl-2-associated x', 'TARGET'), (52, 'bax', 'TARGET'), (55, 'toll-like receptors 4', 'TARGET'), (61, 'tlr4', 'TARGET'), (64, 'p38 mitogen-activated protein kinase', 'TARGET'), (71, 'mapk', 'TARGET')]\n",
      "entities: ('fgl2', 'apoptosis') --> predicted relation: Inhibit\n",
      "entities: ('fgl2', 'tumour necrosis factor') --> predicted relation: Reduce\n",
      "entities: ('fgl2', 'apoptosis') --> predicted relation: Inhibit\n",
      "entities: ('fgl2', 'tumour necrosis factor') --> predicted relation: Reduce\n",
      "entities: ('fgl2', 'b-cell lymphoma-2') --> predicted relation: Decrease\n",
      "entities: ('fgl2', 'bcl2') --> predicted relation: Decrease\n"
     ]
    }
   ],
   "source": [
    "for i in range(len(texts)):\n",
    "    print('\\n\\n\\n')\n",
    "    print('Text : ', texts[i])\n",
    "    preds = re.get_predictions(texts[i], threshold=0.4, disable_ner=True, ents=spans[i])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2214be05-ed2c-496f-9638-2fb2963dd591",
   "metadata": {},
   "source": [
    "### Try with simple sentences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "4bb33feb-3d0e-4ed2-8dfe-89bb9272731b",
   "metadata": {},
   "outputs": [],
   "source": [
    "texts_test = ['A1 inhibits B1', \n",
    "              'A1 inhibited B1', \n",
    "              'A1 has been inhibited by B1', \n",
    "              'A1 does not inhibit B1', \n",
    "              'A1 and B1 and not related', \n",
    "              'A1 is inhibited by B1', \n",
    "              'A1 and B1 does not share an inhibition relation',\n",
    "              'A1 and B1 share an inhibition relation',\n",
    "              \"A1 isn't inhibited by B1\", \n",
    "              'A1 inhibits B1 and B2', \n",
    "              'A1 is inhibited by B1 and inhibits B2', \n",
    "              'A1 inhibits B1 and not B2', \n",
    "              'A1 inhibits B1 and reduces B2', \n",
    "              'A1 does not inhibit B1 but it reduces B2']\n",
    "\n",
    "spans_test = []\n",
    "for t in texts_test:\n",
    "    try:\n",
    "        spans_test.append([(t.index('A1'), t.index('A1') + 2, 'TARGET'), \n",
    "                        (t.index('B1'), t.index('B1') + 2, 'TARGET'), \n",
    "                          (t.index('B2'), t.index('B2') + 2, 'TARGET')])\n",
    "    except ValueError:\n",
    "        spans_test.append([(t.index('A1'), t.index('A1') + 2, 'TARGET'), \n",
    "                 (t.index('B1'), t.index('B1') + 2, 'TARGET')])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "e2a280a2-0f48-43b1-83f9-6eb36f80c400",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Text :  A1 inhibits B1\n",
      "spans: [(0, 'A1', 'TARGET'), (2, 'B1', 'TARGET')]\n",
      "entities: ('A1', 'B1') --> predicted relation: Inhibit\n",
      "\n",
      "\n",
      "Text :  A1 inhibited B1\n",
      "spans: [(0, 'A1', 'TARGET'), (2, 'B1', 'TARGET')]\n",
      "entities: ('A1', 'B1') --> predicted relation: Inhibit\n",
      "\n",
      "\n",
      "Text :  A1 has been inhibited by B1\n",
      "spans: [(0, 'A1', 'TARGET'), (5, 'B1', 'TARGET')]\n",
      "entities: ('B1', 'A1') --> predicted relation: Inhibit\n",
      "\n",
      "\n",
      "Text :  A1 does not inhibit B1\n",
      "spans: [(0, 'A1', 'TARGET'), (4, 'B1', 'TARGET')]\n",
      "entities: ('A1', 'B1') --> predicted relation: Inhibit\n",
      "\n",
      "\n",
      "Text :  A1 and B1 and not related\n",
      "spans: [(0, 'A1', 'TARGET'), (2, 'B1', 'TARGET')]\n",
      "\n",
      "\n",
      "Text :  A1 is inhibited by B1\n",
      "spans: [(0, 'A1', 'TARGET'), (4, 'B1', 'TARGET')]\n",
      "entities: ('B1', 'A1') --> predicted relation: Inhibit\n",
      "\n",
      "\n",
      "Text :  A1 and B1 does not share an inhibition relation\n",
      "spans: [(0, 'A1', 'TARGET'), (2, 'B1', 'TARGET')]\n",
      "\n",
      "\n",
      "Text :  A1 and B1 share an inhibition relation\n",
      "spans: [(0, 'A1', 'TARGET'), (2, 'B1', 'TARGET')]\n",
      "\n",
      "\n",
      "Text :  A1 isn't inhibited by B1\n",
      "spans: [(0, 'A1', 'TARGET'), (5, 'B1', 'TARGET')]\n",
      "entities: ('B1', 'A1') --> predicted relation: Inhibit\n",
      "\n",
      "\n",
      "Text :  A1 inhibits B1 and B2\n",
      "spans: [(0, 'A1', 'TARGET'), (2, 'B1', 'TARGET'), (4, 'B2', 'TARGET')]\n",
      "entities: ('A1', 'B1') --> predicted relation: Inhibit\n",
      "entities: ('A1', 'B2') --> predicted relation: Inhibit\n",
      "\n",
      "\n",
      "Text :  A1 is inhibited by B1 and inhibits B2\n",
      "spans: [(0, 'A1', 'TARGET'), (4, 'B1', 'TARGET'), (7, 'B2', 'TARGET')]\n",
      "entities: ('A1', 'B2') --> predicted relation: Inhibit\n",
      "entities: ('B1', 'B2') --> predicted relation: Inhibit\n",
      "\n",
      "\n",
      "Text :  A1 inhibits B1 and not B2\n",
      "spans: [(0, 'A1', 'TARGET'), (2, 'B1', 'TARGET'), (5, 'B2', 'TARGET')]\n",
      "entities: ('A1', 'B1') --> predicted relation: Inhibit\n",
      "entities: ('A1', 'B2') --> predicted relation: Inhibit\n",
      "\n",
      "\n",
      "Text :  A1 inhibits B1 and reduces B2\n",
      "spans: [(0, 'A1', 'TARGET'), (2, 'B1', 'TARGET'), (5, 'B2', 'TARGET')]\n",
      "entities: ('A1', 'B1') --> predicted relation: Inhibit\n",
      "entities: ('A1', 'B2') --> predicted relation: Reduce\n",
      "\n",
      "\n",
      "Text :  A1 does not inhibit B1 but it reduces B2\n",
      "spans: [(0, 'A1', 'TARGET'), (4, 'B1', 'TARGET'), (8, 'B2', 'TARGET')]\n",
      "entities: ('A1', 'B1') --> predicted relation: Inhibit\n",
      "entities: ('A1', 'B2') --> predicted relation: Reduce\n"
     ]
    }
   ],
   "source": [
    "for i in range(len(texts_test)):\n",
    "    print('\\n')\n",
    "    print('Text : ', texts_test[i])\n",
    "    preds = re.get_predictions(texts_test[i], \n",
    "                       threshold=0.5, \n",
    "                       disable_ner=True, \n",
    "                       ents=spans_test[i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "796df8ca-b5e2-4651-a0a8-7324149090b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "spans_text = ['keratin', 'myosin']\n",
    "text = f\"\"\"\n",
    "{spans_text[0]} is also expressed in human heart and can associate with {spans_text[1]} to suppress \n",
    "its current amplitude and slow the deactivation gating process\n",
    "\"\"\"\n",
    "spans = [(text.index(spans_text[0]), text.index(spans_text[0])+len(spans_text[0]), 'TARGET'), \n",
    "         (text.index(spans_text[1]), text.index(spans_text[1])+len(spans_text[1]), 'TARGET')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "498f65f5-0495-407c-82a5-a9b02084687e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "spans: [(1, 'keratin', 'TARGET'), (12, 'myosin', 'TARGET')]\n",
      "entities: ('keratin', 'myosin') --> predicted relation: Bind\n"
     ]
    }
   ],
   "source": [
    "re.get_predictions(text, 0.5, True, spans)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "f6c1b8d3-680b-42a9-a9ae-583a87a357dd",
   "metadata": {},
   "outputs": [],
   "source": [
    "annotations_train = read_jsonl('RE/assets/annotations_train.jsonl')\n",
    "annotations_test = read_jsonl('RE/assets/annotations_test.jsonl')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "15004436-641c-44a2-92bd-6ca37d4bb899",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "environment": {
   "kernel": "spacy_gpu",
   "name": "common-cu110.m95",
   "type": "gcloud",
   "uri": "gcr.io/deeplearning-platform-release/base-cu110:m95"
  },
  "kernelspec": {
   "display_name": "spacy_gpu",
   "language": "python",
   "name": "spacy_gpu"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
